examples/smoke-test/main.rs

```diff
@@ -48,7 +47,31 @@ fn get_and_write(agent: &ureq::Agent, url: &str) {
     }
 }
 
-fn get_many(urls: Vec<String>, simultaneous_fetches: usize) -> Result<()> {
+fn get_response(agent: &ureq::Agent, url: &str) -> result::Result<Response, ureq::Error> {
+    let fetch = || agent.get(url).call();
+    let mut result = fetch();
+    for _ in 1..4 {
+        let err = match result {
+            Ok(r) => return Ok(r),
+            Err(e) => e,
+        };
+        match err {
+            ureq::Error::Status(429, r) => {
+                let retry: Option<u64> = r.header("retry-after").and_then(|h| h.parse().ok());
+                eprintln!("429 for {}, retrying in {}", r.status(), r.get_url());
+                thread::sleep(Duration::from_secs(retry.unwrap_or(5)));
+                eprintln!("error status {} body {}", r.status(), r.into_string()?);
+            }
+            ureq::Error::Status(_, r) => return Err(r.into()),
+            ureq::Error::Transport(e) => return Err(e.into()),
+        };
```

Comment:
I think this might be more succinctly expressed as:

```rust
if let ureq::Error::Status(429, r) = err {
 // ... retry
} else {
  return Err(err);
}
```